### ISSUE TITLE
In the title, it says risk-conscious. Do you have a REF for your implementation of risk-aware optimization?

### ISSUE THREAD

**Issue by sgbaird** [janosh/thermo#18]

**Comment by janosh** [comment]
We tried several methods of uncertainty estimation for ML models:

- [dropout](https://github.com/janosh/thermo/blob/main/thermo/bnn/torch_dropout.py) [[ref](https://arxiv.org/abs/1506.02142)]
- [multiplicative normalizing flows](https://github.com/janosh/torch-mnf) (worked best overall in our testing) [[ref](https://arxiv.org/abs/1703.01961)]
- ensemble statistics for random forest implemented in a [custom `RandomForestRegressor`](https://github.com/janosh/thermo/blob/main/thermo/rf.py#L54-L121) that sublasses the `scikit-learn` implementation (worked second best likely due to small data set size) [see refs in docstring]
- [Hamiltonian Monte Carlo](https://github.com/janosh/thermo/blob/main/thermo/bnn/hmc.py) (based on [TFP](https://www.tensorflow.org/probability)) (did not seem to scale to the numbers of parameters of even small NNs)
- [Gaussian Process](https://github.com/janosh/thermo/blob/main/thermo/gp.py) (couldn't find a smoothness setting that handled our data well)

The other part that included risk-management in our candidate selection pipeline was to try to decorrelate candidate materials using a branch-and-bound optimization algorithm:

https://github.com/janosh/thermo/blob/812b7f8320ebf8286512dc5a45847ee072a7697b/notebooks/screen/mnf_magpie/mnf_magpie_screen.py#L304-L335
